### PR TITLE
Pets no longer count toward party level

### PR DESCRIPTION
**Pull request by Anexes**
It was possible for low-level pets to reduce party level enough to increase gains and reduce losses from combat. With this change, low-level pets in combat are nothing more than low-level pets. Party level will be the same at the end of combat as it was at the beginning of combat, which prevents any abuse of the xp gain/loss mechanics. (For combat, at least.)

### CODE REVIEW

**Comment by Anexes** [comment]
Nope, it's fine. I thought pets wouldn't be given xp, but they're still in party.players, they just aren't considered for party.level or party.score.

File: src/plugins/party/party.js [hunk 1]
```diff
@@ -23,11 +23,11 @@ export class Party {
   }
 
   get score() {
-    return _.sum(_.map(this.players, 'itemScore')) / this.players.length;
+    return _.sum(_.map(_.filter(this.players, (player) => { return player.isPlayer(); }), 'itemScore')) / this.players.length;
```
Comment (seiyria) [inline src/plugins/party/party.js]: this is not a function. change it to be:

`player => player.isPlayer`

in both places.

**Comment by Anexes** [comment]
Like that?
`_.filter(this.players, player => return player.isPlayer)`

File: src/plugins/party/party.js (hunk 1 above)
Comment (Anexes) [inline src/plugins/party/party.js]: For future reference, what's not a function? isPlayer?

**Comment by seiyria** [comment]
yes, isPlayer is not a function, it's a getter.

**Comment by KennethShen** [comment]
Party score / level calculation isn't quite fixed yet. 

For example, in a party of 4 players, all level 50, the expected average level is 200 / 4 = 50. This is fine. 

If the players all summon 4 pets, the party's total level still stays as 200. However, `this.players.length` will be 8, so the party's average level is now 200/8 = 25.

This affects both the [winners](https://github.com/IdleLands/IdleLands/blob/9ebb91f57db2f15880f9cc82dd60554d4754244c/src/plugins/combat/battle.js#L236) and the [losers](https://github.com/IdleLands/IdleLands/blob/9ebb91f57db2f15880f9cc82dd60554d4754244c/src/plugins/combat/battle.js#L263), as every pet summoned brings their "average level" down.

**Pull request merged**